app/helpers/sessions_helper.rb

```diff
@@ -3,15 +3,44 @@ def log_in user
     session[:user_id] = user.id
   end
 
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
+
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+    return if session_authenticate session[:user_id]
+
+    cookie_authenticate cookies.signed[:user_id]
+  end
+
+  def session_authenticate user_id
+    return if @current_user
+
+    @current_user = User.find_by id: user_id
```

Comment:
```suggestion
    @current_user ||= User.find_by id: user_id
```

Reply:
Khi xử lí như vậy làm rubocop cảnh báo: **Naming/MemoizedInstanceVariableName: Memoized variable @current_user does not match method name session_authenticate. Use @session_authenticate instead.**
Em có thể xử lí như này được không ạ:
` @current_user || @current_user = User.find_by(id: user_id)`